addons/web/static/src/js/views/pivot/pivot_model.js

```diff
@@ -531,539 +455,683 @@ var PivotModel = AbstractModel.extend({
+            domain: groupDomain,
+            fields: measureSpecs,
+            groupBy: groupBy,
+            lazy: false,
+        }).then(function (subGroups) {
+            return {
+                group: group,
+                subGroups: subGroups,
+                rowGroupBy: rowGroupBy,
+                colGroupBy: colGroupBy
+            };
+        });
     },
-    _getCellValue: function (id1, id2) {
-        if ((id1 in this.data.cells) && (id2 in this.data.cells[id1])) {
-            return this.data.cells[id1][id2];
-        }
-        if (id2 in this.data.cells) return this.data.cells[id2][id1];
+    /**
+     * Returns the group sanitized values.
+     *
+     * @private
+     * @param  {Object} group
+     * @param  {string[]} groupBys
+     * @returns {Array}
+     */
+    _getGroupValues: function (group, groupBys) {
+        var self = this;
+        return groupBys.map(function (groupBy) {
+            return self._sanitizeValue(group[groupBy]);
+        });
     },
     /**
-     * @param {any} value
-     * @param {any} root
-     * @param {any} i
-     * @param {any} j
-     * @param {any} parent
-     * @returns {Object}
+     * Returns the leaf counts of each group inside the given tree.
+     *
+     * @private
+     * @param {Object} tree
+     * @returns {Object} keys are group ids
      */
-    _getHeader: function (value, root, i, j, parent) {
-        var path;
-        var total = _t("Total");
-        if (parent) {
-            path = parent.path.concat(value.slice(i,j));
+    _getLeafCounts: function (tree) {
+        var self = this;
+        var leafCounts = {};
+        var leafCount;
+        if (_.isEmpty(tree.directSubTrees)) {
+            leafCount = 1;
         } else {
-            path = [total].concat(value.slice(i,j));
+            leafCount = _.values(tree.directSubTrees).reduce(
+                function (acc, subTree) {
+                    var subLeafCounts = self._getLeafCounts(subTree);
+                    _.extend(leafCounts, subLeafCounts);
+                    return acc + leafCounts[JSON.stringify(subTree.root.values)];
+                },
+                0
+            );
         }
-        return this._findPathInTree(root, path);
+
+        leafCounts[JSON.stringify(tree.root.values)] = leafCount;
+        return leafCounts;
     },
     /**
-     * @private @static
-     * @param {any} header
-     * @returns {integer}
+     * Returns the group sanitized measure values for the measures in
+     * fieldNames (that migth contain '__count', not really a fieldName).
+     *
+     * @private
+     * @param  {Object} group
+     * @param  {string[]} groupBys
+     * @returns {Array}
      */
-    _getHeaderDepth: function (header) {
-        var depth = 1;
-        this._traverseTree(header, function (hdr) {
-            depth = Math.max(depth, hdr.path.length);
-        });
-        return depth;
+    _getMeasurements: function (group, fieldNames) {
+        return fieldNames.reduce(
```

Comment:
(For RCO:)
As soon as alias returned by read_group are or the form field_name:group_operator,
we can use instead of fieldNames the function this._getMeasureSpecs() to reduce.